src/platform/datapath_epoll.c

```diff
@@ -2069,25 +2071,47 @@ CxPlatSocketSendInternal(
     QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
     CXPLAT_SOCKET_CONTEXT* SocketContext = NULL;
     CXPLAT_DATAPATH_PROC_CONTEXT* ProcContext = NULL;
-    ssize_t SentByteCount = 0;
     QUIC_ADDR MappedRemoteAddress = {0};
     struct cmsghdr *CMsg = NULL;
     struct in_pktinfo *PktInfo = NULL;
     struct in6_pktinfo *PktInfo6 = NULL;
     BOOLEAN SendPending = FALSE;
     uint32_t ProcNumber;
+    size_t TotalMessagesCount;
+
+    CXPLAT_DBG_ASSERT(Socket != NULL && RemoteAddress != NULL && SendData != NULL);
```

Comment:
Worth asserting that `SendData->SentMessagesCount` is <= than max?

Reply:
Actually, now that I think of it, does it make sense for this to be called with Count == Max?

Reply:
No, this should always be called with < max.